README.md

```diff
@@ -51,73 +51,74 @@ classes
-  
+
+1. Create a new class that extends `Base_Block` ( `P4GBKS\Blocks\Base_Block` ) inside directory _classes/blocks_. The class name should follow naming convention, for example **Blockname** and its file name should be class-**blockname**.php.
+
+1. Implement its parent's class abstract method. In block's **constructor** you need to define the block's details (fields, schema) using `register_block_type` and in method **prepare_data()** you need to prepare the data which will be used for rendering.
+
+1. Create the template file that will be used to render your block inside directory _templates/blocks_. If the name of the file is **blockname**.twig then
+you need to set the BLOCK_NAME constant as **'blockname'** It also works with html templates. Just add 'php' as the 3rd argument of the block() method.
+
 1. Add your new class name to the array inside Loader's ( `P4GBKS\Loader` ) constructor.
- 
-1. Create a new folder inside _react-blocks/src/blocks_ named after your block **Blockname** (first letter capital - rest lowercase). Create two new files inside that folder named **Blockname.js**  and **BlocknameBlock.js**.  
- 
+
+1. Create a new folder inside _react-blocks/src/blocks_ named after your block **Blockname** (first letter capital - rest lowercase). Create two new files inside that folder named **Blockname.js**  and **BlocknameBlock.js**.
+
 	 **BlocknameBlock.js** should be a class that uses wordpress [registerBlockType](https://developer.wordpress.org/block-editor/developers/block-api/block-registration/) to define the block's attributes, schema and `edit()` function.
- `edit()` function should return a react component that will be used for rendering the block in the editor. 
+ `edit()` function should return a react component that will be used for rendering the block in the editor.
 `save()` function should return null as we use server side rendering currently.
 
-	**Blockname.js** should be a class that defines a react component that implements `renderEdit()` and `render()`.
-`renderEdit()` should be used to render the block in the editor
-`render()` should be used to render the block's preview in the editor using [ServerSideRender](https://developer.wordpress.org/block-editor/components/server-side-render/) and Preview components.
+	**Blockname.js** should be a class that defines a React component that implements `renderEdit()` and `renderView()`.
+`renderEdit()` should be used to render the block in the editor, to define editor-specific things as sidebar options, in-place edit components, and so on. `renderView()` will be used both in the editor and in the frontend site to render the block's contents, as we are rendering blocks using React in the frontend too.
+
+To learn more details about the render logic, refer to the [blocks page in Planet4's Gitbook](https://app.gitbook.com/@greenpeace/s/planet4/tech/blocks/).
```

Comment:
This line stopped the current list (because of indentation ?), so the next 2 items appear as a new one (with [number 1 and 2 displayed](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/tree/planet-4793) instead of 6 and 7)

Reply:
Nice catch! You actually checked the file with formatting. Nice. 💯